src/main/java/com/flowingcode/vaadin/addons/carousel/Carousel.java

```diff
@@ -181,9 +182,16 @@ public void movePos(int slide) {
   // EVENTS
   @DomEvent("position-changed")
   public static class SlideChangeEvent extends ComponentEvent<Carousel> {
-    public SlideChangeEvent(Carousel source, boolean fromClient) {
+	private String position;  
+	  
+    public SlideChangeEvent(Carousel source, boolean fromClient, @EventData("event.detail.value") String position) {
       super(source, true);
     }
+
+	public String getPosition() {
+		return position;
```

Comment:
Code is unformatted (use two spaces instead of a tab). 
Apply the code formatter in your IDE or re-run the maven goal before committing.